webdriver-spec.html

```diff
@@ -6546,185 +6571,331 @@
+              y</var>.</li>
 
-    <p>The <dfn>pointerCancel</dfn> command:
-      <ol>
-        <li><p>If the <a>current top-level browsing context</a> is <a>no longer open</a>
-          , return <a>error</a> with <a>error code</a> <a>no such window</a>.
+          <li><p>Let <var>current x</var> equal the <code>x</code>
+              property of <var>input state</var>.</li>
 
-        <li><p><a>Handle any user prompts</a> and return its value if it is an
-          <a>error</a>.
-        <li><p class=issue>to be completed
-      </ol>
-  </section><!-- /Pointer Actions -->
+          <li><p>Let <var>current y</var> equal the <code>y</code>
+              property of <var>input state</var>.</li>
 
-  <!-- </REMOVE> -->
+          <li><p>If <var>x</var> is not equal to <var>current x</var>
+              or <var>y</var> is not equal to <var>current y</var>, run
+              the following steps:</p>
 
+            <ol>
+              <li><p>Let <var>buttons</var> be equal to input
+                  state's <code>buttons</code> property.</li>
+
+              <li><p><a href="perform-actions">Perform
+                    implementation-specific action dispatch steps</a>
+                    equivalent to moving the pointer with
+                    id <var>source id</var> having
+                    type <var>pointerType</var> from viewport x
+                    coordinate <var>current x</var>, viewport y
+                    coordinate <var>y</var> to viewport x
+                    coordinate <var>x</var> and viewport y
+                    coordinate <var>y</var>, with
+                    buttons <var>buttons</var> depressed, in
+                    accordance with the requirements of [[!UI-EVENTS]]
+                    and [[!POINTER-EVENTS]], with <var>primary</var>
+                    determining whether the pointer is regarded as a
+                    primary pointer and thus generates legacy mouse
+                    events. Type specific properties for the pointer
+                    that are not exposed through the webdriver API
+                    must be set to the default value specified for
+                    hardware that doesn't support that property. In
+                    the case where the <var>pointerType</var>
+                    is <code>"pen"</code> or <code>"touch"</code>,
+                    and <var>buttons</var> is empty, this may be a
+                    no-op. For a pointer of type <code>"mouse"</code>,
+                    this will always produce events including at least
+                    a <code>pointerMove</code> event.</li>
+
+              <li><p>Let <var>input state</var>'s <code>x</code>
+                  property equal <var>x</var> and <code>y</code> property
+                  equal <var>y</var>.</li>
+            </ol>
+          </li>
+
+          <li><p>If <var>last</var> is true, return.
+
+        <li><p><a>Asynchronously wait</a> for an implementation
```

Comment:
It may help implementors to know _why_ there's a wait here. What are we waiting for?